tests/libgstc/c/test_libgstc_pipeline_verbose.c

```diff
@@ -0,0 +1,174 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+#include <gst/check/gstcheck.h>
+#include <string.h>
+
+#include "libgstc.h"
+#include "libgstc_socket.h"
+#include "libgstc_assert.h"
+#include "libgstc_json.h"
+
+/* Test Fixture */
+static gchar _request[512];
+static GstClient *_client;
+
+static void
+setup (void)
+{
+  const gchar *address = "";
+  unsigned int port = 0;
+  unsigned long wait_time = 5;
+  int keep_connection_open = 0;
+
+  gstc_client_new (address, port, wait_time, keep_connection_open, &_client);
+}
+
+static void
+teardown (void)
+{
+  gstc_client_free (_client);
+}
+
+/* Mock implementation of a socket */
+typedef struct _GstcSocket
+{
+} GstcSocket;
+
+GstcSocket _socket;
+
+GstcStatus
+gstc_socket_new (const char *address, const unsigned int port,
+    const int keep_connection_open, GstcSocket ** out)
+{
+  *out = &_socket;
+
+  return GSTC_OK;
+}
+
+void
+gstc_socket_free (GstcSocket * socket)
+{
+}
+
+GstcStatus
+gstc_socket_send (GstcSocket * socket, const gchar * request, gchar ** response,
+    const int timeout)
+{
+  *response = malloc (1);
+
+  memcpy (_request, request, strlen (request));
+
+  return GSTC_OK;
+}
+
+GstcStatus
+gstc_json_get_int (const gchar * json, const gchar * name, gint * out)
+{
+  return *out = GSTC_OK;
+}
+
+GstcStatus
+gstc_json_is_null (const gchar * json, const gchar * name, gint * out)
+{
+  *out = 0;
+  return GSTC_OK;
+}
+
+GstcStatus
+gstc_json_get_child_char_array (const char *json, const char *parent_name,
+    const char *array_name, const char *element_name, char **out[],
+    int *array_lenght)
+{
+  gstc_assert_and_ret_val (NULL != json, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != parent_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != array_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != element_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != out, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != array_lenght, GSTC_NULL_ARGUMENT);
+  return GSTC_OK;
+}
+
+GstcStatus
+gstc_json_child_string (const char *json, const char *parent_name,
+    const char *data_name, char **out)
+{
+  gstc_assert_and_ret_val (NULL != json, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != parent_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != data_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != out, GSTC_NULL_ARGUMENT);
+
+  return GSTC_OK;
+}
+
+GST_START_TEST (test_pipeline_verbose_success)
+{
+  GstcStatus ret;
+  const gchar *pipeline_name = "pipe";
+  const gint value = 1;
```

Comment:
We need a similar test for `false`

Reply:
Yes, now we have:
_test_pipeline_verbose_true_success
test_pipeline_verbose_false_success_